### PR TITLE
Remove `key` field from `provider.Manifest` in the k8s plugin

### PR DESCRIPTION
**Pull request by Warashi**
**What this PR does**:

- remove `key` field from `provider.Manifest` of k8s plugin, and make it in `Key()` method
- add test to ensure application config's namespace is used when applying the manifests
- refactor the tests to reduce duplicated codes

**Why we need it**:

- To reduce modification after the creation of the Manifest instance

**Which issue(s) this PR fixes**:

Part of #4980 

**Does this PR introduce a user-facing change?**: No

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: pkg/app/pipedv1/plugin/kubernetes/deployment/server_test.go [hunk 1]
```diff
@@ -168,3 +191,67 @@ func TestDeploymentService_executeK8sSyncStage(t *testing.T) {
...
+	cfg, err := os.ReadFile(filepath.Join(examplesDir(), "kubernetes", "simple", "app.pipecd.yaml"))
+	require.NoError(t, err)
+
+	// decode and override the autoCreateNamespace and namespace
+	spec, err := config.DecodeYAML[*kubeConfigPkg.KubernetesApplicationSpec](cfg)
+	require.NoError(t, err)
+	spec.Spec.Input.AutoCreateNamespace = true
+	spec.Spec.Input.Namespace = "test-namespace"
+	cfg, err = yaml.Marshal(spec)
+	require.NoError(t, err)
```
Comment (Warashi) [inline pkg/app/pipedv1/plugin/kubernetes/deployment/server_test.go:208]: override example config to ensure these fields are used when applying the resources

File: pkg/app/pipedv1/plugin/kubernetes/deployment/server_test.go [hunk 2]
```diff
@@ -168,3 +191,67 @@ func TestDeploymentService_executeK8sSyncStage(t *testing.T) {
...
+	// initialize plugin config and dynamic client for assertions with envtest
+	pluginCfg, dynamicClient := setupTestPluginConfigAndDynamicClient(t)
+
+	svc := NewDeploymentService(pluginCfg, zaptest.NewLogger(t), testRegistry, logpersistertest.NewTestLogPersister(t))
+	resp, err := svc.ExecuteStage(ctx, req)
+
+	require.NoError(t, err)
+	assert.Equal(t, model.StageStatus_STAGE_SUCCESS.String(), resp.GetStatus().String())
+
+	deployment, err := dynamicClient.Resource(schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"}).Namespace("test-namespace").Get(context.Background(), "simple", metav1.GetOptions{})
```
Comment (Warashi) [inline pkg/app/pipedv1/plugin/kubernetes/deployment/server_test.go:246]: check the resource in the namespace `test-namespace`

**Review by ffjlabo — DISMISSED** [review]
👍

**Comment by Warashi** [comment]
I applied #5450 changes to this PR.

@ffjlabo Sorry, please re-review this PR.

**Review by t-kikuc — APPROVED** [review]

**Review by ffjlabo — APPROVED** [review]

**Pull request merged**